researchstudio-sat/webofneeds
# Feat agreements rest api

Different evaluations concerning highlevel protocols are now available as a Web API in the owner-webapp
How to test (with FireFox):

# Test 1

* Engage in a conversation with the debugbot
* by enabling RDF mode and viewing the linked data for one of your messages in the conversation, you'll find your connection URI (use the object of `msg:hasSender`). Copy that to your clipboard for later.
* Issue the ' propose'  command in the chat
* open the network tab of your **firefox** browser and select one of the `/owner/rest/linked-data/...` requests.
* click on the *Headers* Tab 
* click on * Edit and Resend* 
* edit the URI to `/owner/rest/highlevel/getProposals?connectionUri=` and paste your connection URI there.
* wait for the result. You should see the *Content* of the proposed message in the Response tab.

# Test 2

Same as Test 1 but with a different, real connectionURI to which the user you're currently logged in with does not have access.

# Test 3
Also log in with another user and try again with the connection URI from test 1. You should not get any results.
